contracts/beyondFramework/dca/portfolios/PercentageAllocationPortfolio.sol

```diff
@@ -63,89 +61,78 @@ contract PercentageAllocationPortfolio is
 
     function _authorizeUpgrade(address) internal override onlyOwner {}
 
-    modifier onlyPortfolio() {
-        bool authorized;
-        for (uint256 i = 0; i < portfolios.length; i++) {
-            if (portfolios[i] == _msgSender()) {
-                authorized = true;
-            }
-        }
-
-        require(authorized, "Unauthorized");
-        _;
-    }
-
-    modifier checkEmergencyExited() {
-        for (uint256 i = 0; i < activeStrategies.length; i++) {
-            if (activeStrategies[i].isEmergencyExited()) {
-                activeStrategies[i].withdrawAllFor(
-                    _msgSender(),
-                    false // this param affects nothing in this case
-                );
-
-                activeStrategies[i] = activeStrategies[
-                    activeStrategies.length - 1
-                ];
-                activeStrategies.pop();
-            }
-        }
-
-        _;
-    }
-
-    function deposit(uint256 amount, uint8)
-        external
-        nonReentrant
-        checkEmergencyExited
-    {
+    function deposit(uint256 amount, uint8) external nonReentrant {
         _deposit(_msgSender(), amount);
     }
 
     function depositFor(
         address sender,
         uint256 amount,
         uint8
-    ) external onlyPortfolio nonReentrant checkEmergencyExited {
+    ) external onlyPortfolio nonReentrant {
         _deposit(sender, amount);
     }
 
     function _deposit(address sender, uint256 amount) private {
+        uint256 activeInvestables = _checkActiveInvestables();
+        if (activeInvestables == 0) {
+            revert PortfolioIsEmergencyExited();
+        }
+
         depositToken.safeTransferFrom(_msgSender(), address(this), amount);
 
-        uint256 perDcaDeposit = amount / activeStrategies.length;
-        for (uint256 i = 0; i < activeStrategies.length; i++) {
+        uint256 perInvestableDeposit = amount / activeInvestables;
+        for (uint256 i = 0; i < investables.length; i++) {
             depositToken.safeApprove(
-                address(activeStrategies[i]),
-                perDcaDeposit
+                address(investables[i].investable),
+                perInvestableDeposit
             );
-            activeStrategies[i].depositFor(
+
+            investables[i].investable.depositFor(
                 sender,
-                perDcaDeposit,
+                perInvestableDeposit,
                 investAmountSplit
             );
         }
     }
 
+    function _checkActiveInvestables()
+        private
+        returns (uint256 activeInvestables)
+    {
+        for (uint256 i = 0; i < investables.length; i++) {
+            if (investables[i].emergencyExited) {
+                continue;
+            }
+
+            if (investables[i].investable.isEmergencyExited()) {
+                investables[i].emergencyExited = true;
```

Comment:
It looks like once an investable set as true, it can never be deposited again. Can't it be revived?